nixos/modules/services/web-apps/nextcloud-whiteboard-server.nix

```diff
@@ -0,0 +1,72 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    types
+    literalExpression
+    ;
+  cfg = config.services.nextcloud-whiteboard-server;
+
+in
+{
+  options.services.nextcloud-whiteboard-server = {
+
+    enable = mkEnableOption "Nextcloud backend server for the Whiteboard app";
+
+    settings = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = ''
+        Settings to configure backend server. Especially the Nextcloud host
+        url has to be set. The required environment variable `JWT_SECRET_KEY`
+        should be set via the secrets option.
+      '';
+      example = literalExpression ''
+        {
+          NEXTCLOUD_URL = "https://nextcloud.example.org";
+        }
+      '';
+    };
+
+    secrets = lib.mkOption {
+      type = with types; listOf path;
```

Comment:
This should really be a `str`.

See https://github.com/NixOS/nixpkgs/pull/326306#discussion_r1675992985 for a little bit of context.